hazelcast/src/main/java/com/hazelcast/wan/WanReplicationPublisher.java

```diff
@@ -49,14 +91,147 @@
+     * cluster and does not refer to WAN replication, e.g. between two clusters.
+     * Invoked when migrating WAN replication data between members in a cluster.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @param event      the replication event
+     * @param namespaces namespaces which will be replicated
+     * @return the WAN event container
+     * @see #processEventContainerReplicationData(int, Object)
+     */
+    default T prepareEventContainerReplicationData(PartitionReplicationEvent event,
+                                                   Collection<ServiceNamespace> namespaces) {
+        return null;
+    }
+
+    /**
+     * Processes the WAN event container received through intra-cluster replication
+     * or migration. This method may completely remove existing WAN events for
+     * the given {@code partitionId} or it may append the given
+     * {@code eventContainer} to the existing events.
+     * Invoked when migrating WAN replication data between members in a cluster.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @param partitionId    partition ID which is being replicated or migrated
+     * @param eventContainer the WAN event container
+     * @see #prepareEventContainerReplicationData(PartitionReplicationEvent, Collection)
+     */
+    default void processEventContainerReplicationData(int partitionId, T eventContainer) {
+    }
+
+    /**
+     * Collect the namespaces of all WAN event containers that should be replicated
+     * by the replication event.
+     * Invoked when migrating WAN replication data between members in a cluster.
+     * NOTE: used only in Hazelcast Enterprise.
      *
-     * @throws WANReplicationQueueFullException if queue capacity has been reached and
-     *                                          {@link
-     *                                          com.hazelcast.config.WanBatchReplicationPublisherConfig#getQueueFullBehavior()}
-     *                                          is set to {@link WANQueueFullBehavior#THROW_EXCEPTION}
+     * @param event      the replication event
+     * @param namespaces the set in which namespaces should be added
      */
-    void checkWanReplicationQueues();
+    default void collectAllServiceNamespaces(PartitionReplicationEvent event,
+                                             Set<ServiceNamespace> namespaces) {
+    }
 
+    /**
+     * Removes all WAN events pending replication and belonging to the provided
+     * service and partition.
+     * If the publisher does not store WAN events, this method is a no-op.
+     * Invoked when migrating WAN replication data between members in a cluster.
+     * NOTE: used only in Hazelcast Enterprise.
+     *
+     * @param serviceName the service name of the WAN events should be removed
+     * @param partitionId the partition ID of the WAN events should be removed
+     */
+    default int removeWanEvents(int partitionId, String serviceName) {
```

Comment:
minor: `removeWanEvents` methods can be placed one after another

Reply:
Got it, will do!